xlive/H2MOD/GUI/imgui_integration/Console/ImGui_ConsoleImpl.cpp

```diff
@@ -144,12 +144,11 @@ int Console::TextEditCallback(ImGuiInputTextCallbackData* data)
 			ConsoleCommand* command = completion_commands[i];
 			memset(&console_data->m_completion_data->CompletionCandidate[i], 0, sizeof(ImGuiTextInputCompletionCandidate));
 			console_data->m_completion_data->CompletionCandidate[i].CompletionText = command->GetName();
-			console_data->m_completion_data->CompletionCandidate[i].CompletionVariable = nullptr;
 			console_data->m_completion_data->CompletionCandidate[i].CompletionDescription = command->GetDescription();
 
-			if (auto * var_command = dynamic_cast<ConsoleVarCommand*>(command))
+			if (command->CommandSetsVariable())
 			{
-				console_data->m_completion_data->CompletionCandidate[i].CompletionVariable = var_command->VarAsStr();
+				command->VarAsStr(console_data->m_completion_data->CompletionCandidate[i].CompletionVariable, 512);
```

Comment:
use NUMBEROF(CompletionVariable) instead of putting the size here

Reply:
done